config/config.inc.php

```diff
@@ -120,6 +120,10 @@
 
 $context = Context::getContext();
 
+/* Get smarty */
```

Comment:
I don't get the reason to move this block?

Reply:
Ok, I will try to list the code chain

In config.inc.php we do a Shop::initialize
https://github.com/PrestaShop/PrestaShop/blob/develop/config/config.inc.php#L125

In Shop::initialize, we get the Shop entity https://github.com/PrestaShop/PrestaShop/blob/develop/classes/shop/Shop.php#L395

In Shop constructor, we call setTheme in which we need the ThemeManagerBuilder

ThemeManagerBuilder service requires LegacyContext

In LegacyContext::getContext, we have

```
$adminController = new AdminController();
$adminController->initShopContext();
```

And in the constructor of AdminController

`$this->context->smarty->setTemplateDir`

So if smarty is not set to the context before Shop::initialize, you will have 
`Error: Call to a member function setTemplateDir() on null
`

That's why I moved this block

Reply:
Finally we dont have to use the Sf service to use ThemeManagerBuilder here

we dont need ProviderFactory nor ThemeValidator

I will instantiate a new ThemeManagerBuilder (with null for these 2 parameters). That will be enough